pkg/kubefed2/join.go

```diff
@@ -450,11 +450,22 @@ func createFederationNamespace(clusterClientset kubeclient.Interface, federation
 		return federationNS, nil
 	}
 
-	_, err := clusterClientset.CoreV1().Namespaces().Create(federationNS)
-	if err != nil && !apierrors.IsAlreadyExists(err) {
-		glog.V(2).Infof("Could not create %s namespace: %v", federationNamespace, err)
+	_, err := clusterClientset.CoreV1().Namespaces().Get(federationNamespace, metav1.GetOptions{})
+	if err != nil && !apierrors.IsNotFound(err) {
+		glog.V(2).Infof("Could not get %s namespace: %v", federationNamespace, err)
 		return nil, err
 	}
+
+	if err == nil {
+		glog.V(2).Infof("Already existing %s namespace", federationNamespace)
```

Comment:
I think we can return here and decrease unnecessary indent for `not found` logic.

Reply:
Done.